test/system_benchmarks.cc

```diff
@@ -0,0 +1,69 @@
+#include <catch2/benchmark/catch_benchmark.hpp>
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/generators/catch_generators.hpp>
+
+#include <celerity.h>
+
+#include "test_utils.h"
+
+using namespace celerity;
+
+template <int Dims>
+class bench_runtime_fixture : public test_utils::runtime_fixture {};
+
+TEMPLATE_TEST_CASE_METHOD_SIG(
+    bench_runtime_fixture, "benchmark independent task pattern with N tasks", "[benchmark][system-benchmarks][indep-tasks]", ((int N), N), 100, 1000, 5000) {
+	constexpr size_t num_tasks = N;
+	constexpr size_t num_repeats = 2;
+	constexpr size_t items_per_task = 256;
+
+#ifndef NDEBUG
+	if(N > 100) { SKIP("Skipping larger-scale benchmark in debug build to save CI time"); }
```

Comment:
Good thinking, but do we want to run benchmarks at all in debug builds?

Reply:
Valid question -- another reason I added this code specifically was that I wanted to run the benchmark in debug for development (and, well, *debugging*) reasons, and the larger sizes would take forever. Of course for that use case you could also manually select the individual versions, but with templated test/benchmark cases that isn't particularly convenient.

Reply:
We're mixing `BENCHMARK`s and `CHECK`s in a single test case here so Catch2's `--skip-benchmarks` will probably fail the test. We could think about running the benchmark binaries with `--benchmark-samples=1` to keep the checks working. But anyway, I'm fine with doing this in future PR together with the necessary adjustments to the DAG benchmarks.